src/main/java/com/amazon/dlic/auth/http/saml/SamlDOMMetadataResolver.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.dlic.auth.http.saml;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensaml.saml.metadata.resolver.impl.DOMMetadataResolver;
+
+import java.io.StringReader;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.xml.sax.InputSource;
+
+public class SamlDOMMetadataResolver extends DOMMetadataResolver {
+    private final static Logger log = LogManager.getLogger(SamlDOMMetadataResolver.class);
+
+    public SamlDOMMetadataResolver(final String idpMetadataBody) throws Exception {
+        super(getMetadataDOM(idpMetadataBody));
+    }
+
+    private static Element getMetadataDOM(final String xmlString) throws Exception {
```

Comment:
Move this method to `HTTPSamlAuthenticator`, declare it to throw `ParserConfigurationException`, `SAXException`, `IOException` and delete `SamlDOMMetadataResolver` class.
```suggestion
    private static Element getMetadataDOM(final String xmlString) throws ParserConfigurationException, SAXException, IOException {
        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
        factory.setNamespaceAware(true);

        DocumentBuilder builder = factory.newDocumentBuilder();
        try {
            Document doc = builder.parse(new InputSource(new StringReader(xmlString)));
            return doc.getDocumentElement();
        } catch (SAXException e) {
            log.error("Error while parsing SAML Metadata Body {}", xmlString, e);
            throw e;
        }
    }
```

Reply:
Any advantage of adding the logic in HTTPSamlAuthenticator vs a class. Thought of using a class as it would be consistent with the other Resolvers

Reply:
There is no difference in functionality between `SamlDOMMetadataResolver` and its base class, so introduction of `SamlDOMMetadataResolver` is not necessary, compared to other `MetadataResolver` classes that implement `fetchMetadata()`.